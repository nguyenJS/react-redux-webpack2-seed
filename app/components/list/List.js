import React, { Component } from 'react';
import Team from './team/Team';

import { loadTeams } from '../../services/list/list.service';

export default class List extends Component {
  constructor(props) {
    super(props);
    this.teams = [];
  }

  componentWillMount() {
    this.teams = loadTeams();
  }

  render() {
    const elements = this.teams.map(team => <Team key={team.id} team={team} />);
    return (
        <ul>{elements}</ul>
    );
  }
}

