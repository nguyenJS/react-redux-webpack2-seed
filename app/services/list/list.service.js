export function loadTeams()	{
	return [
		{
			id: "team-1",
			name: "Manchester United",
			creationYear: 1890,
		},
		{
			id: "team-2",
			name: "Manchester City",
			creationYear: 1891,
		},
		{
			id: "team-3",
			name: "Arsenal",
			creationYear: 1892,
		},
		{
			id: "team-4",
			name: "Chelsea",
			creationYear: 1893,
		},
		{
			id: "team-5",
			name: "Liverpool",
			creationYear: 1894,
		}
	];
};
