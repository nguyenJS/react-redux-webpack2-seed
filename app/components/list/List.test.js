import React from 'react';
import { shallow } from 'enzyme';

import List from './List';
import * as listService from '../../services/list/list.service';

describe('List Component', () => {
  it('should render the component', () => {
    // when
    listService.loadTeams = jest.fn().mockReturnValueOnce([{
      id: "team-1",
      name: "Manchester United",
      creationYear: 1890,
    }]);

    const wrapper = shallow(
        <List />
    );
    // then
    expect(wrapper.root.node).toMatchSnapshot();
  });
});

