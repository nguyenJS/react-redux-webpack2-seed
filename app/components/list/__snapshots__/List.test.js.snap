// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component should render the component 1`] = `
<ul>
  <Team
    team={
      Object {
        "creationYear": 1890,
        "id": "team-1",
        "name": "Manchester United",
      }
    }
  />
</ul>
`;
